basics/getting_started/plugin_compatibility.md

```diff
@@ -2,80 +2,102 @@
+| `com.intellij.modules.androidstudio` | Android SDK Platform, Build Tools, Platform Tools, SDK Tools | Android Studio                                                          |
+| `com.intellij.modules.appcode`       | CocoaPods, Core Data Objects, Device & Simulator Support  | AppCode                                                                    |
+| `com.intellij.modules.cidr.lang`     | **C, C++, Objective-C/C++** language PSI Model, Swift/Objective-C Interaction, Inspections, Intentions, Completion, Refactoring, Test Framework  | AppCode, CLion |
+| `com.intellij.modules.cidr.debugger` | Debugger Watches, Evaluations, Breakpoints, Inline Debugging  | AppCode, CLion, RubyMine                                               |
+| `com.intellij.modules.clion`         | CMake, Profiler, Embedded Development, Remote Development, Remote Debug, Disassembly | CLion                                           |
+| `com.intellij.modules.database`      | **SQL** language PSI Model, Inspections, Completion, Refactoring, Queries | IntelliJ IDEA Ultimate Edition, DataGrip, GoLand, PhpStorm, PyCharm, Rider, RubyMine |
+| `com.intellij.modules.go`            | **Go** language PSI Model, Inspections, Intentions, Completion, Refactoring, Test Framework  | GoLand                                  |
+| `com.intellij.modules.python`        | **Python** language PSI Model, Inspections, Intentions, Completion, Refactoring, Test Framework  | PyCharm                             |
+| `com.intellij.modules.rider`         | Connection to **ReSharper** Process in Background   | Rider                                                                            |
+| `com.intellij.modules.ruby`          | **Ruby** language PSI Model, Inspections, Intentions, Completion, Refactoring, Test Framework  | RubyMine                              |
+| `com.jetbrains.php`                  | **PHP** language PSI Model, Inspections, Intentions, Completion, Refactoring, Test Framework | PhpStorm (built-in plugin)              |
+| `com.intellij.modules.webstorm`      | **Web** languages PSI Models, Inspections, Intentions, Completion, Refactoring, Test Framework  | WebStorm                             |
+
+
+## Verifying Dependency
+Before marking a plugin project as _dependent only on modules in a target product_ in addition to `com.intellij.modules.lang`, verify the plugin isn't implicitly dependent on any APIs that are specific to IntelliJ IDEA. 
+To verify a plugin project's independence, create an SDK pointing to an installation of the intended target IntelliJ Platform-based product, e.g., PhpStorm, rather than IntelliJ IDEA. 
+Use the same development version of the IntelliJ platform as the targeted product. 
+Additional product-specific information about developing for the IntelliJ Platform is documented in Part VIII.
```

Comment:
was section "Optional dependencies" removed accidentally? I'd like to keep it

Reply:
I could, but it duplicates the content in [Plugin Dependencies](https://www.jetbrains.org/intellij/sdk/docs/basics/plugin_structure/plugin_dependencies.html#optional-plugin-dependencies), which IMO is a richer explanation. This is explained in the Part I / Part II section on lines 34/35 of this page.

Or are you suggesting we combine the two pages?

Reply:
you're right, let's strip it here then